src/components/ChannelList/ChannelList.tsx

```diff
@@ -1,69 +1,298 @@
+  onChannelTruncated?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Array<Channel<At, Ch, Co, Ev, Me, Re, Us>>>
+    >,
+    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
+  ) => void;
+  /** Function that overrides default behavior when channel gets updated */
+  onChannelUpdated?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Array<Channel<At, Ch, Co, Ev, Me, Re, Us>>>
+    >,
+    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
+  ) => void;
+  /** Function that overrides default behavior when channel becomes visible */
+  onChannelVisible?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Array<Channel<At, Ch, Co, Ev, Me, Re, Us>>>
+    >,
+    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
+  ) => void;
+  /** Function that overrides default behavior when new message is received on channel that is not being watched */
+  onMessageNew?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Array<Channel<At, Ch, Co, Ev, Me, Re, Us>>>
+    >,
+    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
+  ) => void;
+  /** Function that overrides default behavior when users gets removed from a channel */
+  onRemovedFromChannel?: (
+    setChannels: React.Dispatch<
+      React.SetStateAction<Array<Channel<At, Ch, Co, Ev, Me, Re, Us>>>
+    >,
+    event: Event<At, Ch, Co, Ev, Me, Re, Us>,
+  ) => void;
+  /** Object containing query options */
+  options?: ChannelOptions;
+  /**
+   * Paginator component for channels. It contains all the pagination logic such as
+   * - fetching next page of results when needed e.g., when scroll reaches the end of list
+   * - UI to display loading indicator when next page is being loaded
+   * - call to action button to trigger loading of next page.
+   *
+   * Available built-in options (also accepts the same props as):
+   *
+   * 1. [LoadMorePaginator](https://github.com/GetStream/stream-chat-react/blob/master/src/components/LoadMore/LoadMorePaginator.tsx)
+   * 2. [InfiniteScrollPaginator](https://github.com/GetStream/stream-chat-react/blob/master/src/components/InfiniteScrollPaginator/InfiniteScrollPaginator.tsx)
+   */
+  Paginator?: React.ComponentType<
+    InfiniteScrollPaginatorProps | LoadMorePaginatorProps
+  >;
+  /**
+   * Available built-in options (also accepts the same props as):
+   *
+   * 1. [ChannelPreviewCompact](https://getstream.github.io/stream-chat-react/#ChannelPreviewCompact) (default)
+   * 2. [ChannelPreviewLastMessage](https://getstream.github.io/stream-chat-react/#ChannelPreviewLastMessage)
+   * 3. [ChannelPreviewMessenger](https://getstream.github.io/stream-chat-react/#ChannelPreviewMessanger)
+   *
+   * The Preview to use, defaults to ChannelPreviewLastMessage
+   */
+  Preview?: React.ComponentType<
+    ChannelPreviewUIComponentProps<At, Ch, Co, Ev, Me, Re, Us>
+  >;
+  /** Last channel will be set as active channel if true, defaults to true */
+  setActiveChannelOnMount?: boolean;
+  showSidebar?: boolean;
```

Comment:
this need a comment?

Reply:
didn't add anything that wasn't already present cause of our ticket in the end to run through everything then